### PR TITLE
Sdk access token file

### PR DESCRIPTION
**Pull request by nickchapman-da**
Support authenticating ledgers in SDK:

- support `--access-token-file` flag in daml-helper ledger/deploy commands:
    - `daml deploy`
    - `daml ledger upload-dar`
    - `daml ledger allocate-parties`
- update doc and help
- add integration test

This PR address github issue https://github.com/digital-asset/daml/issues/3158

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags, if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs [hunk 1]
```diff
@@ -786,21 +788,35 @@ runStart
                 Nothing -> f sandboxPh
                 Just jsonApiPort -> withJsonApi sandboxPort jsonApiPort args f
 
-data HostAndPortFlags = HostAndPortFlags { hostM :: Maybe String, portM :: Maybe Int }
+data LedgerFlags = LedgerFlags
+  { hostM :: Maybe String
+  , portM :: Maybe Int
+  , jwtFileM :: Maybe String
```
Comment (cocreature) [inline daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs]:
```suggestion
  , jwtFileM :: Maybe FilePath
```

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs [hunk 2]
```diff
@@ -786,21 +788,35 @@ runStart
...
 
-getHostAndPortDefaults :: HostAndPortFlags -> IO HostAndPort
-getHostAndPortDefaults HostAndPortFlags{hostM,portM} = do
+getJwtFromFile :: Maybe String -> IO (Maybe Jwt)
+getJwtFromFile jwtFileM = do
+  case jwtFileM of
+    Nothing -> return Nothing
+    Just jwtFile -> do
+      contents <- readFileUTF8 jwtFile
+      let jwt = either error id $ Jwt.tryCreateFromString contents
```
Comment (cocreature) [inline daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs]:
```suggestion
      jwt <-either fail pure $ Jwt.tryCreateFromString contents
```

File: language-support/hs/bindings/BUILD.bazel [hunk 3]
```diff
@@ -9,6 +9,7 @@ da_haskell_library(
     name = "hs-ledger",
     srcs = glob(["src/**/*.hs"]),
     hackage_deps = [
+        "MissingH",
```
Comment (cocreature) [inline language-support/hs/bindings/BUILD.bazel]: Adding a dep on `MissingH` just for `strip` seems a bit excessive. If you do want to strip, I would prefer to use `Text.strip` or just inline an implementation using `takeWhile`.

**Comment by cocreature** [comment]
#3609 has been merged so this needs to be rebased.

**Comment by associahedron** [comment]
Awesome, thanks! 👍

**Pull request merged**